### ISSUE TITLE
Labeling open bugs by version and find means to check if they are still reproducible

### ISSUE THREAD

**Issue by andrii-korotkov-verkada** [argoproj/argo-cd#20753]
# Summary

There are labels from `version:2.11` to `version:2:14` (for changes from master for now) as well as `verison:EOL` to track the most recent version where things reproduce. This can be helpful for managing and prioritizing bugs.

# Motivation

There are ~1600 open bugs which need some managing and prioritization. Some might get fixed as a class by upgrading, e.g. 2.13 includes perf optimizations for getting app tree, which can solve a whole class of bugs associated with a slow refresh.
The elephant in the room is that there are not enough community resources at the moment to tackle everything. But it's important to start somewhere.

# Proposal

The naive approach is to go over issues, find the most recent version mentioned, tag it and ask people to upgrade if it has reached EOL. This may work for people how filed only a handful of issues, but doesn't work for people who filed a lot and can't do repros for every one of them.
A slow approach is to try reproducing every issue, but this doesn't scale well in the current situation.
A middle ground can be only doing this for people with <5 filed issues or when there are no clear repro steps, as mentioned by @jsoref, and having separate queues of issues for people who filed a lot or there are clear steps to repro and take a slow approach there.

**Comment by andrii-korotkov-verkada** [comment]
Based on the discussion at contributors meeting people overwhelmingly feeling like closing those stale issues and offering to re-open if necessary. Probably same applies to PRs.

**Comment by andrii-korotkov-verkada** [comment]
@crenshaw-dev, @reggie-k, @leoluz, did you have a chance to discuss this on the maintainers meeting? What's the outcome?

**Comment by crenshaw-dev** [comment]
I provided a vigorous (minority) defense of not auto-closing issues or PRs. The consensus was that we should probably automate adding stale labels but that all issue/PR closing should be manual.

**Comment by andrii-korotkov-verkada** [comment]
Something like: we add a stale label if there's been no updates in 1 year and do this every 3 months to not be aggressive. Then we close stale issues every 3 months before running the script again unless there are some exceptions about those issues, e.g. high number of likes.

**Comment by jsoref** [comment]
Humans should review tickets, it's really valuable. Every month someone can look at the 10 oldest tickets and try to decide if they're relevant/need information and provide that human touch.

**Comment by andrii-korotkov-verkada** [comment]
Yeah, I agree, I do review old tickets as well as most starred ones. But there are just too many.

**Comment by jsoref** [comment]
For those, you can ask the reporter. If they don't answer within 3 months (this tolerates people being on summer holidays), it's ok to close.

**Comment by andrii-korotkov-verkada** [comment]
Looks like there's no good way to close the tickets. I'll close as complete due to 1st part (labeling by version) seemingly working out. But the 2nd part I consider an unsolvable problem.
